modules/alloy/src/main/ts/ephox/alloy/ui/schema/SplitSlidingToolbarSchema.ts

```diff
@@ -58,7 +59,11 @@ const parts: () => PartType.PartTypeAdt[] = Fun.constant([
             },
             onStartGrow: (comp) => {
               AlloyParts.getPart(comp, detail, 'overflow-button').each(Toggling.on);
-            }
+              detail.onToggled(true);
```

Comment:
It's a bit inconsistent that we directly call `onToggled` here for Sliding, but for Floating, we rely on the Toggling behaviour's handler being defined. The Toggling behaviour for the button for sliding is just below (about 20-30 lines below this). Could we just put `onToggled` into that config like we do for Floating? Is there a reason for the different approach between the two?

Reply:
Because now we've kind of combined approaches. Due to the animation aspects of sliding we have got this: 

* onShrunk calls `Toggling.off` but doesn't fire handler
* onStartGrow calls `Toggling.on` *and* does fire handler
* onStartShrink fires handler but doesn't call `Toggling.off` yet. 

So the handler isn't mapped to Toggling, which I guess is why you didn't use the Toggling config approach.

If this extra complexity is all created by the need to make sure the handler fires as soon as the close is issued, rather than after the animation is finished, maybe we should revisit that requirement. By decoupling Toggling and the handler like this, we're arguably making the interactions even harder to understand.

Reply:
- [ ] if we go with this approach, let's put thorough comments in the code as to why. Otherwise, the next developer who looks at this code is probably just going to change it to make it seem more consistent.

And I think I'm unconvinced that we want to introduce so much complexity just so the event doesn't wait for the sliding animation to complete before firing ... but I might be in the minority on that.

Reply:
Or if we stick to this, then maybe we should decouple the other one as well. Not sure.

Reply:
@ephox-mogran Added comments to the `SplitSlidingToolbarSchema` [here](https://github.com/tinymce/tinymce/pull/8238/commits/d16d3768492e9013664463f92ee70b1b68e024d4).

I would prefer to stick to the current implementation, because the behaviour is much more clear for the end-user of the `onToggled` callback. In the current case it calls immediately after the button click.

Otherwise, if we will add `onToggled` to the `Toggling` behaviour, the `ToggleToolbarEvent` will be called in the beginning of the toolbar's opening, but in the end of the toolbar's closing. That's extremely confusing. 

Would discuss it with the team, but, again, personally I would prefer the current implementation.

Reply:
Maybe it is also a good idea to add `OpenToggleToolbarDrawer` and `CloseToggleToolbarDrawer` events, which correspond to the `Toggling` behaviour of the `SplitSlidingToolbarSchema`.

Reply:
Yeah, it's messy either way. I wonder if the problem here was more that we initially made those calls animation dependent. And we probably did that because our actual use case needed it to be animation dependent. I'm happy to defer to whatever @TheSpyder , @ltrouton , and @spocke think, as the official TinyMCE reviewers. As long as we have an explanation for why we've done it this way in the code and the commit message, I don't mind so much.

Reply:
I would've expected the event to not fire until the animation is complete but I can see valid reasons either way.

I'm fine with just explaining it as Morgan suggested.

Reply:
@ephox-mogran @TheSpyder I've discussed when to call the callback with the team and we've arrived to a conclusion to call `onToggled` **after** the animation is completed. Updated the code accordingly [here](https://github.com/tinymce/tinymce/pull/8238/commits/8d8ddca29583d055293eda91a0b852e8a1d9c00c). What do you think?

Reply:
So I've looked at the recent changes, and I have a few more comments. It's probably easier to discuss these via a call, but I don't know if that will suit you time-wise. Anyway..

* When declaring handlers in alloy, try not to remove the component from the arguments. It's so hard to get built components in alloy. Your current code may not need it, but someone else's code might. Always get into the habit of all handlers having component as their first argument
* The SlidingToolbar now takes a configuration setting `onToggled` that it doesn't appear to use. I don't think it's necessary any more.
* However, instead of making an `onToggled` setting in FloatingToolbar, it might be an idea to mimic what Sliding has and have a onOpened and onClosed for consistency.
* The test you've added doesn't test the state being passed through the handler, and also only tests the button. You'd want something that checks that the FloatingToolbar is correctly configuring its button also. I'm not sure if there's an existing test.

It looks good otherwise.

Reply:
@ephox-mogran 
1. Updated all callbacks with the `comp` variable containing `AlloyComponent`;
2. Removed `onToggled` handler from the `SlidingToolbar`;
3. Renamed `onToggled` handler to `onOpened` and `onClosed` in `FloatingToolbar`;
4. Updated `FloatingButtonTest` to check the state variable;
5. Updated `FloatingToolbarTest` with `onOpened` and `onClosed` callbacks.